dev/lint-python

```diff
@@ -26,30 +32,45 @@ PYLINT_REPORT_PATH="$SPARK_ROOT_DIR/dev/pylint-report.txt"
 PYLINT_INSTALL_INFO="$SPARK_ROOT_DIR/dev/pylint-info.txt"
 SPHINXBUILD=${SPHINXBUILD:=sphinx-build}
 SPHINX_REPORT_PATH="$SPARK_ROOT_DIR/dev/sphinx-report.txt"
+VIRTUALENV_EXEC="virtualenv"
+
+# It is a good practice to upgrade pip to the latest version, however, sometimes it might break
+# something. Let freeze this version here. We cannot put it into requirements.txt since the
+# execution of this requirement files might depends on the pip version.
+PIP_VERSION="8.1.2"
 
 cd "$SPARK_ROOT_DIR"
 
+DEBUG=false
+if [[ $1 == "--debug" || $1 == "-d" ]]; then
+    DEBUG=true
+fi
+
+
+if [[ ! -z $VIRTUAL_ENV  || ! -z $CONDA_PREFIX ]]; then
+    echo "We already are inside a virtual environment: $VIRTUAL_ENV$CONDA_PREFIX. Using it"
+else
+  if hash conda 2>/dev/null; then
+    echo "Not inside a virtual environment. Creating conda virtual env"
+    conda create -y -n $VIRTUAL_ENV_NAME python
+    source activate $VIRTUAL_ENV_NAME
+    echo "To activate this environment, use: 'source activate $VIRTUAL_ENV_NAME'"
+  else
+    echo "Not inside a virtual environment. Jumping into one in: $VIRTUAL_ENV_DIR"
+    $VIRTUALENV_EXEC $VIRTUAL_ENV_DIR || exit 1
+    source $VIRTUAL_ENV_DIR/bin/activate
+    echo "To activate this environment, use: 'source $SPARK_ROOT_DIR/$VIRTUAL_ENV_DIR/bin/activate'"
+  fi
+fi
+
+pip install --upgrade "pip==$PIP_VERSION"
```

Comment:
you should not use `pip` with anaconda, because they might collide. Use `pip` as a last resort if there is no `conda` equivalent. you you use `conda create -n new environment --file req.txt`, and I think this has a different syntax than `requirements.txt`